### PR TITLE
Hotfix and misc. changes

### PR DESCRIPTION
**Pull request by AluminumAlman**
## Changes

- Anyone parrying no longer crashes the game under certain conditions (caused by partial penetrations or deflections on parry);
- Improved description of toughness stat;
- Improved description of parry chance stat;
- Improved description of critical chance stat;
- Improved description of melee weapon counter parry bonus stat;
- Modified the toughness autopatcher to exclude ranged weapons from the x2 toughness modifier of blunt weapons.

## Testing

- [x] Compiles without warnings;
- [x] Game runs without errors;
- [ ] (For compatibility patches) ...with and without patched mod loaded;
- [ ] Playtested a colony (specify how long);
- [x] Anyone parrying no longer crashes the game (testing done on save [tester.zip](https://github.com/CombatExtended-Continued/CombatExtended/files/8539689/tester.zip) by attacking the spear pawn with the knife pawn).

### CODE REVIEW

**Comment by AluminumAlman** [comment]
Closing until I'm confident with the PR.

**Comment by AluminumAlman** [comment]
Reopening after new commits.

File: Source/CombatExtended/CombatExtended/ArmorUtilityCE.cs [hunk 1]
```diff
@@ -502,14 +502,15 @@ public static void ApplyParryDamage(DamageInfo dinfo, Thing parryThing)
                 }
 
                 var penAmount = dinfo.ArmorPenetrationInt; //GetPenetrationValue(dinfo);
+                //TryPenetrateArmor(dinfo.Def, parryThingArmor, ref penAmount, ref dmgAmount, parryThing);
                 if (TryPenetrateArmor(dinfo.Def, parryThingArmor, ref penAmount, ref dmgAmount, parryThing))
                 {
                     // Partially penetrating sharp attacks.
-                    if (dinfo.Def.armorCategory == DamageArmorCategoryDefOf.Sharp)
+                    if (dinfo.Def.armorCategory == DamageArmorCategoryDefOf.Sharp && dinfo.Amount > dmgAmount)
                         ApplyParryDamage(GetDeflectDamageInfo(dinfo, dinfo.HitPart, ref dmgAmount, ref penAmount, true), parryThing);
```
Comment (perkinslr) [inline Source/CombatExtended/CombatExtended/ArmorUtilityCE.cs:510]: This should probably be 
```c#
if (dinfo.Def.armorCategory == DamageArmorCategoryDefOf.Sharp) {
    ApplyParryDamage(GetDeflectDamageInfo(dinfo, dinfo.HitPart, ref dmgAmount, ref penAmount, (dinfo.Amount > dmgAmount)), parryThing);
```

File: Source/CombatExtended/CombatExtended/ArmorUtilityCE.cs [hunk 2]
```diff
@@ -502,14 +502,15 @@ public static void ApplyParryDamage(DamageInfo dinfo, Thing parryThing)
                 }
 
                 var penAmount = dinfo.ArmorPenetrationInt; //GetPenetrationValue(dinfo);
+                //TryPenetrateArmor(dinfo.Def, parryThingArmor, ref penAmount, ref dmgAmount, parryThing);
```
Comment (perkinslr) [inline Source/CombatExtended/CombatExtended/ArmorUtilityCE.cs:505]: Dead end comment should go away.

**Review by perkinslr — DISMISSED** [review]

**Review by ItsSamGray — APPROVED** [review]

**Review by perkinslr — APPROVED** [review]

**Pull request merged**